include/zephyr/bluetooth/gatt.h

```diff
@@ -1368,7 +1368,12 @@ typedef uint8_t (*bt_gatt_discover_func_t)(struct bt_conn *conn,
 enum {
 	/** Discover Primary Services. */
 	BT_GATT_DISCOVER_PRIMARY,
-	/** Discover Secondary Services. */
+	/** Discover Secondary Services.
+	*
+	* There is no procedure for discovering secondary
+	* services. This is handled by the Find Included Services
+	* sub-procedure as part of Relationship Discovery.
+	*/
```

Comment:
There is indeed no spec defined procedures for it, but it is still possible to discover secondary services. 

In Zephyr we use either the `gatt_find_type` or `gatt_read_group` when discovering secondary services: 

```c
	switch (params->type) {
	case BT_GATT_DISCOVER_PRIMARY:
	case BT_GATT_DISCOVER_SECONDARY:
		if (params->uuid) {
			return gatt_find_type(conn, params);
		}
		return gatt_read_group(conn, params);
```

Reply:
You can try this with the BT shell and use `gatt discover-secondary`

Reply:
> There is indeed no spec defined procedures for it, but it is still possible to discover secondary services.
> 
> In Zephyr we use either the `gatt_find_type` or `gatt_read_group` when discovering secondary services:
> 
> ```c
> 	switch (params->type) {
> 	case BT_GATT_DISCOVER_PRIMARY:
> 	case BT_GATT_DISCOVER_SECONDARY:
> 		if (params->uuid) {
> 			return gatt_find_type(conn, params);
> 		}
> 		return gatt_read_group(conn, params);
> ```

The original suggestion was that secondary discovery is not supported as per the BT spec Vol 3 Part G 2.6.2 but reading the spec again it just says there no specified procedure, so I guess this is a non issue. 
Although reading the comment again there was another part to it which I had left out.

"The API document should explicitly state that a secondary service is only intended to be included from a primary service
or another secondary service or other higher layer specification" Do you think this is ok to mention?

Reply:
> "The API document should explicitly state that a secondary service is only intended to be included from a primary service or another secondary service or other higher layer specification" Do you think this is ok to mention?

Yes, something like that, however whether or not a secondary service is included is not really relevant to the discovery procedure here. We can discover and use a secondary however we like, regardless of knowing the actual primary service. 

The GATT spec does require a secondary service to be included by a primary service (i.e. it cannot function exclusively on it's own). 

So I am unsure whether we need to add anything for the discovery function.